core/trino-main/src/test/java/io/trino/operator/TestStreamingExchangeClientBuffer.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.DataSize;
+import io.trino.execution.StageId;
+import io.trino.execution.TaskId;
+import io.trino.execution.buffer.PageCodecMarker;
+import io.trino.execution.buffer.SerializedPage;
+import io.trino.spi.QueryId;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestStreamingExchangeClientBuffer
+{
+    private List<TaskId> tasks;
+    private List<SerializedPage> pages;
+
+    @BeforeClass
+    public void beforeClass()
+    {
+        StageId stageId = new StageId(new QueryId("query"), 0);
+        tasks = ImmutableList.of(new TaskId(stageId, 0, 0), new TaskId(stageId, 1, 0));
```

Comment:
nit: extract constants for `TASK_0`, `TASK_1`, `PAGE_1`, `PAGE_2`, `PAGE_3`.
Use those here and in assertions below.

Also it looks like all the fields can be constants.